src/utils/helpers.py

```diff
@@ -1,12 +1,33 @@
 import json
 import logging
+import shutil
 from pathlib import Path
 
 import yaml
 
 PATH_SPARK_LOG = "/tmp/logs/spark.log"
 
 
+def clear_directory(dir: Path | str) -> None:
```

Comment:
I switched back to this helper function because `shutil.rmtree` also deletes the directory itself, and I wanted to preserve the directory structure and `.gitkeep` file of `target/initial` and `target/final`.